### PR TITLE
Fix DeiT doc tets

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Fixes failing doctest for DeiT cause by copy-pasta from the PT model. 

The predicted class is determinstic with the set seed. However, as RNG for pytorch and tensorflow are different, it's a different class than in the pytorch Deit doctest. 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by ydshieh** [comment]
From the description & the change, it seems to me that Deit has some random OP(s) used in the inference. Is this the case (and if yes, could you share which line that random op is 🙏 ) Thank you.

**Comment by amyeroberts** [comment]
Sure :) `DeiTForImageClassification` loaded from the official facebook hub weights will have its head randomly initialized. You can't see it directly in this diff, but there's a comment above on line L928 mentioning this.

**Review by LysandreJik — APPROVED** [review]

**Review by ydshieh — COMMENTED** [review]
Thank you! 

(BTW, this is not the random OP I mentioned earlier - it is just the missing head as you mentioned)

**Review by ydshieh — APPROVED** [review]
